jsonnet/prometheus-adapter.jsonnet

```diff
@@ -0,0 +1,133 @@
+local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
+local service = k.core.v1.service;
+local deployment = k.apps.v1beta2.deployment;
+local container = deployment.mixin.spec.template.spec.containersType;
+local volume = deployment.mixin.spec.template.spec.volumesType;
+local configmap = k.core.v1.configMap;
+local containerPort = container.portsType;
+local containerVolumeMount = container.volumeMountsType;
+local tmpVolumeName = 'volume-directive-shadow';
+local tlsVolumeName = 'kube-state-metrics-tls';
+
+{
+  prometheusAdapter+:: {
+    local tlsVolumeName = 'prometheus-adapter-tls',
+    local tlsPath = '/etc/tls/private',
+
+    local prometheusAdapterPrometheusConfig = 'prometheus-adapter-prometheus-config',
+    local prometheusAdapterPrometheusConfigPath = '/etc/prometheus-config',
+
+    local servingCertsCABundle = 'serving-certs-ca-bundle',
+    local servingCertsCABundleDirectory = 'ssl/certs',
+    local servingCertsCABundleFileName = 'service-ca.crt',
+    local servingCertsCABundleMountPath = '/etc/%s' % servingCertsCABundleDirectory,
+
+    apiService+:
+      {
+        metadata+: {
+          annotations+: {
+            'service.alpha.openshift.io/inject-cabundle': 'true',
+          },
+        },
+        spec+: {
+          insecureSkipTLSVerify: false,
+        },
+      },
+
+    service+:
+      {
+        metadata+: {
+          annotations+: {
+            'service.alpha.openshift.io/serving-cert-secret-name': tlsVolumeName,
+          },
+        },
+        spec+: {
+          type: 'ClusterIP',
+        },
+      },
+
+    deployment+:
+      {
+        spec+: {
+          template+: {
+            spec+: {
+              containers:
+                std.map(
+                  function(c)
+                    if c.name == 'prometheus-adapter' then
+                      c
+                      {
+                        args+: [
+                          // '--prometheus-ca-file=%s/%s' % [servingCertsCABundleMountPath, servingCertsCABundleFileName],
+                          // '--prometheus-token-file=/var/run/secrets/kubernetes.io/serviceaccount/token',
```

Comment:
@metalmatze I guess these are left-overs? If so, let's get rid of commented code.

Reply:
I left that in there for now until we now how the adapter moves on. 😉

Reply:
Maybe add the discussion issue to the comment